sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelHandler.java

```diff
@@ -114,9 +113,14 @@ protected void initChannel(final Channel channel) {
         }
 
         pipeline.addFirst(
-            // TODO (DANOBLE) Log an issue with netty
-            // Initialize sslHandler with jdkCompatibilityMode = true for openssl context.
-            new SslHandler(this.config.sslContext().newEngine(channel.alloc())),
+            // Initialize sslHandler with jdkCompatibilityMode = true for OpenSSL context
+            // TODO Log an issue with netty for clarification on the design of this constructor and the
+            //  semantic differences between the JDK and OpenSSL implementations
+            new SslHandler(this.config.sslContext().newEngine(channel.alloc())));
+
+        final long idleConnectionTimerResolutionInNanos = config.idleConnectionTimerResolutionInNanos();
+
+        pipeline.addFirst(
```

Comment:
this adds IdleStateHandler before SSLHandler. Is is intentional?

The old code had IdleStateHandler after SSLHandler.

Reply:
I believe SSLHandler should be the very first one in the pipeline, no ?

Reply:
good catch, I do not thinkthat is the intent here, reverted back.

Reply:
@kushagraThapar  yup, SSLHandler is the very first one in the pipeline today, but after the change, IdleStateHandler will be the first one, so did not think that is the intent. Has reverted the change back.

Thanks~